tools/topology/topology2/cavs-sdw.conf

```diff
@@ -103,159 +104,7 @@ IncludeByKey.SDW_DMIC {
 "1" "platform/intel/sdw-dmic-generic.conf"
 }
 
-#
-# List of all DAIs
-#
-#ALH Index: 0, Direction: duplex
-Object.Dai.ALH [
-	{
-		dai_index 0
-		id 		$SDW_JACK_OUT_BE_ID
-		direction	"playback"
-		name		$SDW_JACK_OUT_STREAM
-		default_hw_conf_id	0
-		rate			48000
-		channels		2
-
-		Object.Base.hw_config.1 {
-			id	0
-			name	"ALH2"
-		}
-	}
-	{
-		dai_index 10
-		id 		$SDW_JACK_IN_BE_ID
-		direction	"capture"
-		name		$SDW_JACK_IN_STREAM
-		default_hw_conf_id	1
-		rate			48000
-		channels		2
-
-		Object.Base.hw_config.1 {
-			id	1
-			name	"ALH3"
-		}
-	}
-]
-
-#
-# Pipeline definitions
-#
-
-# Pipeline ID:1 PCM ID: 0
-Object.Pipeline {
-	host-copier-gain-mixin-playback [
-		{
-			index 0
-
-		Object.Widget.copier.1 {
-			stream_name "volume playback 0"
-		}
-		Object.Widget.gain.1 {
-			Object.Control.mixer.1 {
-				name '1 Playback Volume 0'
-				}
-			}
-		}
-	]
-
-	mixout-gain-dai-copier-playback [
-		{
-			index 1
-
-			Object.Widget.copier.1 {
-				stream_name $SDW_JACK_OUT_STREAM
-				dai_type "ALH"
-				copier_type "ALH"
-				node_type $ALH_LINK_OUTPUT_CLASS
-			}
-			Object.Widget.gain.1 {
-				Object.Control.mixer.1 {
-					name '2 Main Playback Volume'
-				}
-			}
-		}
-	]
-
-	host-gateway-capture [
-		{
-			index 10
-
-			Object.Widget.copier.1.stream_name	"Passthrough Capture 0"
-			Object.Widget.copier.1.Object.Base.audio_format.1 {
-				# 32/32 -> 16/16 bits conversion is done here
-				in_bit_depth	32
-				in_valid_bit_depth	32
-			}
-		}
-	]
-
-	highpass-capture-be [
-		{
-			direction	"capture"
-			index 11
-			copier_type "ALH"
-
-			Object.Widget.copier.1 {
-				stream_name	$SDW_JACK_IN_STREAM
-				dai_type	"ALH"
-				copier_type	"ALH"
-				type		"dai_out"
-				node_type $ALH_LINK_INPUT_CLASS
-			}
-			Object.Widget.eqiir.1 {
-				Object.Control.bytes."1" {
-					name '4 Main capture Iir Eq'
-				}
-			}
-		}
-	]
+IncludeByKey.SDW_JACK {
```

Comment:
assuming there's no Jack, do we need the deep buffer? or does it need to move into sdw-jack.conf as well?

Reply:
> assuming there's no Jack, do we need the deep buffer? or does it need to move into sdw-jack.conf as well?

I thought about this. But consider that we may add deep buffer support for amp someday, and it won't compile if we include deep-buffer.conf in both sdw-jack-generic.conf and sdw-amp-generic.conf. So, I decide to include deep-buffer.conf in cavs-sdw.conf and set DEEPBUFFER_FW_DMA_MS in the cmake file.
BTW, setting DEEPBUFFER_FW_DMA_MS=0 will still include deep-buffer.conf. We need to set DEEPBUFFER_FW_DMA_MS to an alphabet, like "N" to remove deep buffer support.

Reply:
Discussed by Pierre that it makes no sense for amp to use deep buffer since the power saved by deep buffer is insignificant compare to power consumption by amp.

Reply:
Thanks @RanderWang @ranj063, moved deep-buffer.conf to sdw-jack-generic.conf now